backpack/core/derivatives/batchnorm_nd.py

```diff
@@ -323,16 +320,3 @@ def _get_free_axes(
         for n in range(self._get_n_axis(module)):
             free_axes.append(index_batch + n + 2)
         return tuple(free_axes)
-
-    @staticmethod
-    def _maybe_warn_no_batch_summation(sum_batch: bool) -> None:
-        """Warn that Jacobians w.r.t. single components are not per-sample gradients.
-
-        Args:
-            sum_batch: Whether to sum out the batch dimension.
-        """
-        if not sum_batch:
-            warn(
-                "BatchNorm batch summation disabled."
-                "This may not compute meaningful quantities"
-            )
```

Comment:
Can you raise this warning only if the module is in train mode?

Reply:
Sure this would be possible.
At the same time, there is also a warning/error raised in the BatchNorm extensions, e.g. [here](https://github.com/f-dangel/backpack/blob/afc94f1d9a1666e56f0cefea24c2288f928f6a56/backpack/extensions/firstorder/batch_grad/batchnorm_nd.py#L17).
Do we want to have both of them?

Reply:
To get a better overview, which extensions do what, you can also look into https://github.com/f-dangel/backpack/pull/206, where all these code lines are refactored into `check_hyperparameters_module_extension`.

Reply:
Ah I understand. Your solution is fine, thanks again for the pointers :)